src/init.c

```diff
@@ -67,6 +67,21 @@ static char shmem_internal_my_hostname[MAXHOSTNAMELEN];
 static char shmem_internal_my_hostname[HOST_NAME_MAX];
 #endif
 
+
+static inline long errchk_atol(char *s)
+{
+    long val;
+    char *e;
+    errno = 0;
+
+    val = strtol(s,&e,0);
+    if(errno != 0) {
+        perror("env var conversion");
+        exit(1);
+    }
```

Comment:
Unfortunately, the strtol error check doesn't do what you are hoping for.  See: http://stackoverflow.com/questions/9445171/strtol-using-errno

The better way to do this is to use sscanf:
```
int nread = sscanf(s, "%ld", &e);
if (nread != 1) {
  shmem_runtime_abort(1, "Error parsing environment variable");
}